include/noise/protocol/util.h

```diff
@@ -30,6 +30,8 @@
 extern "C" {
 #endif
 
+int noise_init();
```

Comment:
Change this to:

int noise_init(void);

C compilers can act weird with K&R prototypes.

I'm not too sure whether noise_init() needs to be in the public API.  Functions like noise_cipherstate_new_by_id(), noise_dhstate_new_by_id(), etc can call it internally, with a boolean flag for "already initialized".